renderer/src/features/mcp-servers/hooks/use-mutation-restart-server.ts

```diff
@@ -178,10 +177,26 @@ export function useMutationRestartServer({
 
       return { previousData }
     },
-    onSuccess: () => {
-      queryClient.invalidateQueries({
-        queryKey: getApiV1BetaWorkloadsQueryKey({ query: { all: true } }),
-      })
+    onSuccess: async () => {
+      // Poll until server running
+      await pollBatchServerStatus(
+        async (names) => {
+          const statusResponses = await Promise.all(
+            names.map((name) =>
+              queryClient.fetchQuery(
+                getApiV1BetaWorkloadsByNameStatusOptions({ path: { name } })
+              )
+            )
+          )
+          return statusResponses.map((response, index) => ({
+            name: names[index],
+            status: response.status || 'unknown',
+          })) as CoreWorkload[]
```

Comment:
I think this is not needed anymore, can you quickly try to remove it, it should just work